In [ ]:
pip install --upgrade pip

In [ ]:
pip install pandas scikit-learn Keras tensorflow

In [ ]:
pip install tensorflow==2.2.0

# Import all dependencies

In [ ]:
import pandas as pd
from keras.models import Sequential
from keras.layers import Dense, Dropout
from keras.wrappers.scikit_learn import KerasClassifier
from sklearn.model_selection import GridSearchCV

# Load datasets -> Dados e Clases

In [ ]:
data_entry = pd.read_csv("dataset/entry_breast.csv")
data_class = pd.read_csv("dataset/exits_breast.csv")

# Configures the model for training

In [ ]:
def create_rede(optimizer, loss, kernel_initializer, activation, neurons):
    _classifier = Sequential()  # Sequential model.
    # Connected NN layer (Input Layer)
    _classifier.add(Dense(units=neurons, activation=activation, kernel_initializer=kernel_initializer, input_dim=30))
    # Dropout
    _classifier.add(Dropout(0.1))
    # Hidden Layer
    _classifier.add(Dense(units=neurons, activation=activation, kernel_initializer=kernel_initializer))
    # Dropout
    _classifier.add(Dropout(0.1))
    # Hidden Layer
    _classifier.add(Dense(units=neurons, activation=activation, kernel_initializer=kernel_initializer))
    # Output Layer
    _classifier.add(Dense(units=1, activation='sigmoid'))
    # Configures the model for training
    _classifier.compile(optimizer=optimizer, loss=loss, metrics=['binary_accuracy'])
    return _classifier

# Implementation of the scikit-learn classifier API for Keras

In [ ]:
classifier = KerasClassifier(build_fn=create_rede)

# Exhaustive search over specified parameter values for an estimator

Parameters:

In [ ]:
parameters = {
    'batch_size': [5, 10, 20, 30],
    'epochs': [50, 75, 100, 120],
    'optimizer': ['adam', 'sgd'],
    'loss': ['binary_crossentropy', 'hinge'],
    'kernel_initializer': ['random_uniform', 'normal'],
    'activation': ['relu', 'tanh'],
    'neurons': [8, 16, 24, 32]
}
# cv = cross-validation generator
grid_search = GridSearchCV(estimator=classifier, param_grid=parameters, scoring='accuracy', cv=5)

Run fit with all sets of parameters

In [ ]:
grid_search_fit = grid_search.fit(data_entry, data_class)

In [18]:
best_parameters = grid_search_fit.best_params_
better_accuracy = grid_search_fit.best_score_

In [19]:
print(best_parameters)
print(better_accuracy)

{'activation': 'relu', 'batch_size': 20, 'epochs': 120, 'kernel_initializer': 'random_uniform', 'loss': 'binary_crossentropy', 'neurons': 16, 'optimizer': 'adam'}
0.9156652693681104


{
    'activation': 'relu', 
    'batch_size': 20, 
    'epochs': 120, 
    'kernel_initializer': 'random_uniform',
    'loss': 'binary_crossentropy',
    'neurons': 16,
    'optimizer': 'adam'
}

0.9156652693681104